src/Security/Authorization/test/AuthorizationMiddlewareTests.cs

```diff
@@ -658,6 +661,27 @@ public async Task IAuthenticateResultFeature_UsesExistingFeatureAndResult_Withou
         Assert.Same(authenticateResult, authenticateResultFeature.AuthenticateResult);
     }
 
+    // Validation for https://github.com/dotnet/aspnetcore/issues/43188
+    [Fact]
+    public async Task WebApplicationBuilder_CanRegisterAuthzMiddlewareWithScopedService()
+    {
+        var builder = WebApplication.CreateBuilder(new WebApplicationOptions()
+        {
+            EnvironmentName = Environments.Development
```

Comment:
Inconsistent scopes on resolved services only get validated in `Development` so we set the environment name explicitly here to validate the broken behavior.